modules/python/test/test_misc.py

```diff
@@ -50,39 +68,538 @@ class Arguments(NewOpenCVTests):
         self.assertEqual(res3, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=2 dims(-1)=1 size(-1)=2x1 type(0)=CV_32SC1 dims(0)=2 size(0)=2x3 type(0)=CV_32SC1")
-        c = np.array([[[1,2],[3,4],[5,6]]], dtype='f')
+        c = np.array([[[1, 2], [3, 4], [5, 6]]], dtype='f')
         res4 = cv.utils.dumpInputArrayOfArrays([c, a, b])
         self.assertEqual(res4, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=3 dims(-1)=1 size(-1)=3x1 type(0)=CV_32FC2 dims(0)=2 size(0)=3x1 type(0)=CV_32FC2")
 
+    def test_parse_to_bool(self):
+        def try_to_convert(value):
+            try:
+                result = cv.utils.dumpBool(value).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, value, type(value).__name__
+                    )
+                )
+            else:
+                return result
+
+        for convertible_true in (True, 1, 64, np.bool(1), np.int8(123),
+                                 np.int16(11), np.int32(2), np.int64(1)):
+            actual = try_to_convert(convertible_true)
+            self.assertEqual('bool: true', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: true' vs Actual: {2}".format(
+                                 convertible_true,
+                                 type(convertible_true).__name__,
+                                 actual
+                             ))
+
+        for convertible_false in (False, 0, np.uint8(0)):
+            actual = try_to_convert(convertible_false)
+            self.assertEqual('bool: false', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: false' vs Actual: {2}".format(
+                                 convertible_false,
+                                 type(convertible_false).__name__,
+                                 actual
+                             ))
+
+        a = np.array([1, 2, 0], dtype=np.bool)
+        for i in range(len(a)):
+            expected = 'Bool: {}'.format(bool(a[i])).lower()
+            actual = try_to_convert(a[i])
+            self.assertEqual(expected, actual,
+                             msg="Can't convert array element {0} of type {1}\n"
+                             "Expected: {2} vs Actual: {3}".format(
+                                 a[i],
+                                 type(a[i]).__name__,
+                                 expected, actual
+                             ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpBool((True, False, False)[0:1])
+
+        for not_convertible in (1.2, np.float(2.3), 's', 'str', (1, 2),
```

Comment:
Perhaps it makes sense to split this test case: `test_parse_to_bool`, `test_not_parse_to_bool`?

Same for other test cases.

Reply:
Sounds good, done.